paths/paths_unix.go

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build !windows
+
+package paths
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"golang.org/x/sys/unix"
+)
+
+func init() {
+	stateFileFunc = stateFileUnix
+}
+
+func stateFileUnix() string {
+	// TODO: use other default paths on other GOOSes probably. This works for Linux.
+	const varLib = "/var/lib/tailscale/tailscaled.state"
+	try := varLib
+	for i := 0; i < 3; i++ { // check writability of the file, /var/lib/tailscale, and /var/lib
+		err := unix.Access(try, unix.O_RDWR)
+		println(fmt.Sprintf("Access(%q) = %v", try, err))
+		if err == nil {
+			return varLib
+		}
+		try = filepath.Dir(try)
+	}
+
+	// TODO: try some $HOME/.tailscale or XDG path? But will it
+	// even work usefully enough as non-root? Probably not. Maybe
+	// best to require it be explicit in that case.
+	return ""
```

Comment:
$STATE_DIRECTORY is what systemd exports when the unit defines a state directory. It'd be nice to support that automagically, but we can also pass in an explicit state in the service config for that case.

Reply:
You want to add that in a follow-up?

Reply:
Up to you. I'm fine just explicitly specifying a statefile for systemd for now, until we have to do it some other way (which may be the case with some advanced fancy sandboxing, but we can cross that when we get to it)